src/components/PageSwitcher/components/EEGEduEvoked/EEGEduEvoked.js

```diff
@@ -0,0 +1,280 @@
+          {"This will ensure that consecutive rows of your output file are not overlapping in time."}
+          {"It will make the live plots appear more choppy."}
+        </p>        
+      </Card.Section>
+      <Stack>
+        <ButtonGroup>
+          <Button 
+            onClick={() => {
+              saveToCSV(Settings);
+              recordPopChange();
+            }}
+            primary={status !== generalTranslations.connect}
+            disabled={status === generalTranslations.connect}
+          > 
+            {'Save to CSV'}  
+          </Button>
+        </ButtonGroup>
+        
+        <Modal
+          open={recordPop}
+          onClose={recordPopChange}
+          title={"Recording Data"}
+        >
+          <Modal.Section>
+           <Card.Section>
+              <P5Wrapper sketch={sketchEvoked} />          
+            </Card.Section>
+            <TextContainer>
+              <p>
+                Your data is currently recording, 
+                once complete it will be downloaded as a .csv file 
+                and can be opened with your favorite spreadsheet program. 
+                Close this window once the download completes.
+              </p>
+            </TextContainer>
+          </Modal.Section>
+        </Modal>
+
+      </Stack>
+    </Card>
+  )
+}
+
+function saveToCSV(Settings) {
+  const numSamplesToSave = 2000;
+  console.log('Saving ' + numSamplesToSave + ' samples...');
+  var localObservable$ = null;
+  const dataToSave = [];
+  window.marker = 0;
+
+  console.log('making ' + Settings.name + ' headers')
+
+  // for each module subscribe to multicast and make header
+  // take one sample from selected observable object for headers
+  localObservable$ = window.multicastEvoked$.pipe(
+  take(1)
+  );
+  //take one sample to get header info
+  localObservable$.subscribe({ 
+  next(x) { 
+    dataToSave.push(
+      "Timestamp (ms),",
+      "Marker,",
+      generateXTics(x.info.samplingRate,x.data[0].length,false).map(function(f) {return "ch0_" + f + "ms"}) + ",", 
+      generateXTics(x.info.samplingRate,x.data[0].length,false).map(function(f) {return "ch1_" + f + "ms"}) + ",", 
+      generateXTics(x.info.samplingRate,x.data[0].length,false).map(function(f) {return "ch2_" + f + "ms"}) + ",", 
+      generateXTics(x.info.samplingRate,x.data[0].length,false).map(function(f) {return "ch3_" + f + "ms"}) + ",", 
+      generateXTics(x.info.samplingRate,x.data[0].length,false).map(function(f) {return "chAux_" + f + "ms"}) + ",", 
+      "info", 
+      "\n"
+    );   
+  }
+  });
+  // put selected observable object into local and start taking samples
+  localObservable$ = window.multicastEvoked$.pipe(
+  take(numSamplesToSave)
+  );
+
+  // now with header in place subscribe to each epoch and log it
+  localObservable$.subscribe({
+    next(x) { 
+      dataToSave.push(Date.now() + "," + window.marker + "," + Object.values(x).join(",") + "\n");
```

Comment:
this adds a column for window.marker, which is assigned false above to initialize (making it actually 3 lines changed), and changed inside the sketch